dal/src/main/java/com/craftsmanship/tfm/dal/grpc/server/GrpcServer.java

```diff
@@ -29,144 +29,144 @@
-
-      responseObserver.onNext(response);
-      responseObserver.onCompleted();
+    public void initialize() {
     }
 
-    @Override
-    public void get(GetItemRequest request, io.grpc.stub.StreamObserver<GetItemResponse> responseObserver) {
-      logger.info("GET RPC CALLED");
-
-      Item item = dataAccess.get(request.getId());
-      GrpcItem grpcItem = getGrpcItemFromItem(item);
-    		                		  
-      GetItemResponse response = GetItemResponse.newBuilder().setItem(grpcItem).build();
-
-      responseObserver.onNext(response);
-      responseObserver.onCompleted();
+    @Component
+    class ItemPersistenceServiceImpl extends ItemPersistenceServiceImplBase {
+
+        @Autowired
+        private DataAccess dataAccess;
+
+        public ItemPersistenceServiceImpl(DataAccess dataAccess) {
+            this.dataAccess = dataAccess;
+        }
+
+        @Override
+        public void create(CreateItemRequest request, io.grpc.stub.StreamObserver<CreateItemResponse> responseObserver) {
+            logger.info("CREATE RPC CALLED");
+            GrpcItem grpcItem = request.getItem();
+            Item item = getItemFromGrpcItem(grpcItem);
+            Item createdItem = dataAccess.create(item);
+
+            GrpcItem grpcItemResponse = getGrpcItemFromItem(createdItem);
+            CreateItemResponse response = CreateItemResponse.newBuilder().setItem(grpcItemResponse).build();
+
+            responseObserver.onNext(response);
+            responseObserver.onCompleted();
+        }
+
+        @Override
+        public void list(Empty request, io.grpc.stub.StreamObserver<ListItemResponse> responseObserver) {
+            logger.info("LIST RPC CALLED");
+
+            Builder responseBuilder = ListItemResponse.newBuilder();
+            for (Item item : dataAccess.list()) {
+                GrpcItem grpcItem = getGrpcItemFromItem(item);
+                responseBuilder.addItem(grpcItem);
+            }
+            ListItemResponse response = responseBuilder.build();
+
+            responseObserver.onNext(response);
+            responseObserver.onCompleted();
+        }
+
+        @Override
+        public void get(GetItemRequest request, io.grpc.stub.StreamObserver<GetItemResponse> responseObserver) {
+            logger.info("GET RPC CALLED");
+
+            Item item = dataAccess.get(request.getId());
+            GrpcItem grpcItem = getGrpcItemFromItem(item);
+
+            GetItemResponse response = GetItemResponse.newBuilder().setItem(grpcItem).build();
+
+            responseObserver.onNext(response);
+            responseObserver.onCompleted();
+        }
+
+        @Override
+        public void update(UpdateItemRequest request, io.grpc.stub.StreamObserver<UpdateItemResponse> responseObserver) {
+            logger.info("UPDATE RPC CALLED");
+
+            // TODO: right now, if the id does not exists, the item is creted, Should we
+            // raise an error?
```

Comment:
I think we should raise an exception, but I think we can live without it by the moment. There are too many exceptions we have to add...